src/components/GuestWelcomeWindow.vue

```diff
@@ -0,0 +1,149 @@
+<!--
+  - @copyright Copyright (c) 2022, Dorra Jaouad <dorra.jaoued1@gmail.com>
+  -
+  - @author Dorra Jaouad <dorra.jaoued1@gmail.com>
+  -
+  - @license AGPL-3.0-or-later
+  -
+  - This program is free software: you can redistribute it and/or modify
+  - it under the terms of the GNU Affero General Public License as
+  - published by the Free Software Foundation, either version 3 of the
+  - License, or (at your option) any later version.
+  -
+  - This program is distributed in the hope that it will be useful,
+  - but WITHOUT ANY WARRANTY; without even the implied warranty of
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+  -
+  -->
+
+<template>
+	<NcModal v-if="showModal"
+		:container="container"
+		:can-close="false"
+		:close-on-click-outside="false"
+		size="small">
+		<div class="modal__content">
+			<div class="conversation-information">
+				<ConversationIcon :item="conversation"
+					:show-user-status="false"
+					:disable-menu="true" />
+				<h2>{{ conversationDisplayName }}</h2>
+			</div>
+			<h3 class="description">
+				{{ conversationDescription }}
+			</h3>
+
+			<h3 class="input-label">
+				{{ t('spreed', 'Enter your name') }}
+			</h3>
+			<SetGuestUsername ref="setGuestUsername"
+				class="guest-user-form"
+				first-welcome
```

Comment:
Maybe don't use this component here, NcTextField will be enough? That means copy-pasting the method, but if you don't use a half of component functionality (NcButton), why do we need it?

Reply:
It means basically copying more than half of the `SetGuestUsername` component if I go with simply NcTextField. It seems more like a duplication .